### DISCUSSION TITLE
When running the User Management Starter SQL script, only the profiles table is created

### DISCUSSION THREAD

**Discussion by eliasljubec** [supabase/supabase#3201] [Questions] [answered]
as the title indicates, only the table "profiles" is created with the "User Management Starter SQL" script, and no other tables, which should not be the case according to the docs.

![image](https://user-images.githubusercontent.com/36228716/133283008-348ac6b9-b0a2-4b03-94a7-75b90c8e6565.png)

I have seen from other users that they have significantly more tables after running the script, as seen in the screenshot.

![image](https://user-images.githubusercontent.com/36228716/133283552-936a490f-0a04-4219-b8cb-d8a701c9ab4e.png)

compared to my tables after the execution of the script..

![image](https://user-images.githubusercontent.com/36228716/133283653-a5e8c366-0707-4990-8a55-438f4ee1eb1d.png)

I also noticed that the created profiles are not saved in the profiles table, it just stays empty.

a solution regarding the problem would be helpful!

**Comment by dshukertjr** [comment]
Hi @eljubec!

What you are seeing seems fine! What you are seeing on the second screenshot with bunch of tables is a bit old, and there was an update to hide all the system tables from that page. Here is what I see when I go to the page now:

<img width="1386" alt="Screen Shot 2021-09-15 at 6 48 18" src="https://user-images.githubusercontent.com/18113850/133338477-05af13bd-e789-4317-8974-ffc398d97c28.png">

All of the tables that you see here are tables created by me, and does not include any tables automatically created by Supabase such as `migrations` or `objects`. 

>I also noticed that the created profiles are not saved in the profiles table, it just stays empty.

Which quick starter guide are you following by the way? Also, when you say "profiles are not saved in the profiles table", do they seemed like they are saved on the client side? When you refresh the page, do you still see the profile that you created?

Could you confirm that you do not see any data when you go look at the table contents from the tables editor either?

<img width="956" alt="Screen Shot 2021-09-15 at 6 54 05" src="https://user-images.githubusercontent.com/18113850/133339083-43df0f6e-5157-4fa0-b0f5-b1fe82dc5581.png">

**Comment by eliasljubec** [comment, marked as answer]
> Which quick starter guide are you following by the way? Also, when you say "profiles are not saved in the profiles table", do they seemed like they are saved on the client side? When you refresh the page, do you still see the profile that you created?

I have followed the React documents. I meant, for example, when I create a user with the supabase.auth.signUp function, when created successfully, I see the user data in the menu - Authentication, but not in the menu - Tables under Profiles, which should be the case to establish the further use of the data such as relationships with other tables.

![image](https://user-images.githubusercontent.com/36228716/133340627-01d75109-4aa1-4253-88db-6d5be580b4e3.png)

![image](https://user-images.githubusercontent.com/36228716/133340652-1a7abdcd-341a-4725-aaa5-c8a661a1dede.png)

**Reply by dshukertjr** [reply]
@eljubec 

Everything that you have described seems to be working as intended! Running only `supabase.auth.signUp()` function will only create a user in the `auth.users` table (this is what you are seeing under `Authentication` menu), and nothing on `public.profiles` table (this is what you are seeing in the tables editor page). 

Have you already implemented [`account_page`](https://supabase.io/docs/guides/with-react#account-page)? In order to save something in `public.profiles` table, you need to save data like this:

```typescript
const updates = {
  id: user.id,
  username,
  website,
  avatar_url,
  updated_at: new Date(),
}

let { error } = await supabase.from('profiles').upsert(updates, {
  returning: 'minimal', // Don't return the value after inserting
})
```

Or another thing you can do is to set up a trigger on the database so that when a new Auth user is created, a row in profiles table is automatically created as well:

```sql
create function public.handle_new_user() 
returns trigger as $$
begin
  insert into public.profiles (id)
  values (new.id);
  return new;
end;
$$ language plpgsql security definer;
create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

You can read more about database triggers with Auth [here](https://github.com/supabase/supabase/discussions/306)!

**Reply by eliasljubec** [reply]
Okay i'm a little confused, why not just have access to the **auth.users** table?  Where is the advantage of having 2 tables with "similar" data? Then I would no longer need to set up a trigger to automatically copy new users from auth.users to my public.profiles table.

But I suspect because of some internal authentication process? I have not found any documentation about why it is the way it is.

So this means when I update a user with the auth.update() function, I will have to first set up a trigger to update my user information in the public.profiles table in order to apply the changes to my public table? And the same procedure (with triggers etc.) to create and store a user in the public.profiles table?

**Reply by MildTomato** [reply]
It could be good practise to treat your profiles table as 'user meta data'. 
You won't need to store anything in there that is already in auth.users like email, provider, created_at etc since you can access those using [auth.user()](https://supabase.io/docs/reference/javascript/auth-user).

You could also treat your profiles table with only upserting data to profiles when/as you need to - so lets say, if in the profiles table you want a column called 'nickname', then you only need to upsert a record when the user wants to create a nickname for themselves in your app - otherwise there's no need to create a record for the user in the table if there's nothing to store in there for them.

> But I suspect because of some internal authentication process? I have not found any documentation about why it is the way it is.

we could possibly do a better job here of explaining what is in this table and why it is not a good idea to use it to store user metadata. Other than the table is already quite dense with very specific auth details for handling different providers, this table (i think..) is also periodically updated so supabase can continue to improve the Auth service, for example, recently we added phone auth.

**Reply by dshukertjr** [reply]
@eljubec 

As MildTomato has said, you could treat `profiles` table as like a table where you store meta data about the user such as name, description, profile image URL or anything that you would like to store there. If you don't have any information that you would like to store as such, you don't need to create a `profiles` table at all. 

You could instead create a table that has foreign key relationship with `auth.users` table via SQL. 
```sql
create table if not exists public.posts (
  id uuid not null primary key DEFAULT uuid_generate_v4 (),
  owner uuid references auth.users on delete cascade not null,
  contnet varchar(320) not null,
  image_url text,
);
```

This seems like is what you are looking for. The quick starter guide has `profiles` table, because it needs to store user meta information such as name or image_url.

**Reply by dshukertjr** [reply]
@eljubec 
From the client, you will be interacting with `auth.users` table though the Auth API's. If you want to get any data of currently signed in user, you can use [`auth.user()`](https://supabase.io/docs/reference/javascript/auth-user). 

But it sounds like you want to retrieve all users' data at once? What is your goal here? What information of users do you want to retrieve and why? `auth.users` table is not designed for that as it is reserved to safely authenticate and authorize users. If you want to load list of user profiles, then you should setup `public.profiles` table to do it.

**Reply by eliasljubec** [reply]
Never mind, got it! Thank you very much! Good reference source: https://supabase-cheatsheet.vercel.app/docs/auth/public-profiles-table
